packages/vuetify/src/components/VIcon/VIcon.ts

```diff
@@ -182,7 +176,7 @@ const VIcon = mixins(
     const icon = this.getIcon()
 
     if (typeof icon === 'string') {
-      if (icon.indexOf('svg-') === 0) {
+      if (/^[mzlhvcsqta]\s*[-+.0-9][^mlhvzcsqta]+/i.test(icon) && /[\dz]$/i.test(icon) && icon.length > 4) {
```

Comment:
This line deserves some comment with explanation

Could be also extracted to a seperate methods and made more readable

Reply:
> Could be also extracted to a seperate method

done